packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -18,6 +18,7 @@ import slash from 'slash';
 import {sync as writeFileAtomic} from 'write-file-atomic';
 import type {Config} from '@jest/types';
 import HasteMap from 'jest-haste-map';
+import {shouldLoadAsEsm} from 'jest-resolve';
```

Comment:
Can you investigate if we change in `ScriptTransformer` is the correct place? I see that this line https://github.com/jestjs/jest/blob/580d4b731db43e130d63ffc22b445fe2f3102c05/packages/jest-runtime/src/index.ts#L719 seems to be more like the root cause of the issue.

Can you investigate further into `jest-runtime` to identify where we should put the fix